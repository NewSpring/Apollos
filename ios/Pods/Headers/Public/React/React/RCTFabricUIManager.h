../../../../../../node_modules/react-native/React/Fabric/RCTFabricUIManager.h