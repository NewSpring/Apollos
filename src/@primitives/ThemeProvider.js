import { PureComponent, Children } from 'react';
import PropTypes from 'prop-types';
import { DEFAULT_THEME, THEME_PROPS } from './constants';


export default class ThemeProvider extends PureComponent {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.shape(THEME_PROPS),
  };

  static defaultProps = {
    children: null,
    theme: DEFAULT_THEME,
  };

  static childContextTypes = {
    theme: PropTypes.shape(THEME_PROPS),
  };

  getChildContext = () => ({
    theme: Object.assign({}, DEFAULT_THEME, this.props.theme),
  })

  render() {
    return Children.only(this.props.children);
  }
}
