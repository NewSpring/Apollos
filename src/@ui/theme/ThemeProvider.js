import { PureComponent, Children } from 'react';
import PropTypes from 'prop-types';
import createTheme, { THEME_PROPS } from './createTheme';


export default class ThemeProvider extends PureComponent {
  static propTypes = {
    children: PropTypes.node,
    themeInput: PropTypes.shape(THEME_PROPS),
  };

  static defaultProps = {
    children: null,
    themeInput: {},
  };

  static childContextTypes = {
    theme: PropTypes.shape(THEME_PROPS),
    themeInput: PropTypes.shape(THEME_PROPS),
  };

  getChildContext = () => ({
    theme: createTheme(this.props.themeInput),
    themeInput: this.props.themeInput,
  })

  render() {
    return Children.only(this.props.children);
  }
}
