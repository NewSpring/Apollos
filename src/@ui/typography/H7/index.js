import { Platform, Text } from 'react-native';
import PropTypes from 'prop-types';
import { compose, pure, setPropTypes } from 'recompose';
import styled from '@ui/styled';
import { withPlaceholder, Typography } from '@ui/Placeholder';

const styles = styled(
  ({ theme }) => ({
    fontSize: theme.helpers.rem(0.778),
    fontFamily: theme.typography.fontFamilySans.regular.default,
    color: theme.colors.text.primary,
    ...Platform.select({
      ios: {
        paddingTop: theme.helpers.rem(0.2),
        paddingBottom: theme.helpers.rem(0.1),
        lineHeight: theme.helpers.verticalRhythm(0.778, 1.014),
      },
      android: {
        paddingTop: theme.helpers.rem(0.1),
        lineHeight: theme.helpers.verticalRhythm(0.778, 1.214),
      },
      web: {
        paddingTop: theme.helpers.rem(0.225),
        paddingBottom: theme.helpers.rem(0.085),
        lineHeight: theme.helpers.verticalRhythm(0.778, 1.05),
        WebkitFontSmoothing: 'antialiased',
        MozOsxFontSmoothing: 'grayscale',
      },
    }),
  }),
  'H7',
);

const H7 = compose(
  setPropTypes({
    isLoading: PropTypes.bool, // display loading placeholder
    ...Text.propTypes,
  }),
  styles,
  withPlaceholder(Typography, { width: '40%' }),
  pure,
)(Text);

export default H7;
