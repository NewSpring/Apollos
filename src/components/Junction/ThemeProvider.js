import { PureComponent, Children } from 'react';
import PropTypes from 'prop-types';
import { DEFAULT_THEME } from './constants';

export default class ThemeProvider extends PureComponent {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.shape({
      primaryColor: PropTypes.string,
      secondaryColor: PropTypes.string,
    }),
  };

  static defaultProps = {
    children: null,
    theme: DEFAULT_THEME,
  };

  static childContextTypes = {
    theme: PropTypes.shape({
      primaryColor: PropTypes.string,
      secondaryColor: PropTypes.string,
    }),
  };

  getChildContext = () => ({
    theme: this.props.theme,
  })

  render() {
    return Children.only(this.props.children);
  }
}
