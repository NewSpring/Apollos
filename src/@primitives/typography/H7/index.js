import React from 'react';
import { Text } from 'react-native';
import { compose, pure, setPropTypes } from 'recompose';
import PropTypes from 'prop-types';
import styled from '@primitives/styled';
import rem from '@utils/remUnit';
import verticalRhythm from '@utils/verticalRhythm';

const enhance = compose(
  pure,
  setPropTypes({
    children: PropTypes.node,
    style: Text.propTypes.style,
  }),
);

const StyledH7 = styled(({ theme }) => ({
  fontSize: rem(0.778, theme),
  fontFamily: theme.fontFamilySans,
  lineHeight: verticalRhythm(0.778, 1.02, theme),
  color: theme.baseFontColor,
}))(Text);

const H7 = enhance(({
  children,
  style: styleProp = {},
  ...otherProps
}) => (
  <StyledH7
    style={styleProp}
    {...otherProps}
  >
    {children}
  </StyledH7>
));

export default H7;
