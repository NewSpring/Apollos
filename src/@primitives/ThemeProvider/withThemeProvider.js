import { mapProps, withContext, compose } from 'recompose';
import PropTypes from 'prop-types';
import { THEME_PROPS } from '@primitives/constants';
import withTheme from '@primitives/withTheme';

// HOC to help with extending theme.
// Use to over-ride theme context for child components
// Does not change or modify props of the component you're enhancing
const withThemeProvider = themeInput => compose(
  mapProps(props => ({ ownProps: props })),
  withTheme(),
  withContext({ theme: PropTypes.shape(THEME_PROPS) }, ({ theme, ownProps }) => {
    let composedTheme = Object.assign({}, theme, ownProps.theme || {});

    let themeInputAsObject = themeInput;
    if (typeof themeInput === 'function') themeInputAsObject = themeInput({ ...ownProps, theme: composedTheme });

    composedTheme = Object.assign({}, composedTheme, themeInputAsObject);

    return ({
      theme: composedTheme,
    });
  }),
  mapProps(({ ownProps }) => ownProps),
);

export default withThemeProvider;
