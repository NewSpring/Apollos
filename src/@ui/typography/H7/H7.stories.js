import React from 'react';

import { storiesOf } from '@storybook/react-native';

import { ThemeProvider } from '@ui/theme';
import H7 from './';

storiesOf('@ui/typography', module)
  .add('H7', () => (
    <ThemeProvider>
      <H7>{'"We may speak about a place where there are no tears, no death, no fear, no night; but those are just the benefits of heaven. The beauty of heaven is seeing God." â€• Max Lucado'}</H7>
    </ThemeProvider>
  ));
