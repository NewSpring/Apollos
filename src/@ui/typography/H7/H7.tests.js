import React from 'react';
import renderer from 'react-test-renderer';
import ThemeProvider from '@ui/ThemeProvider';
import H7 from './';

describe('the H7 component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <H7>Default H7 text</H7>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should accept and render passed in styles', () => {
    const salmon = { color: 'salmon' };
    const tree = renderer.create(
      <ThemeProvider>
        <H7 style={salmon}>Salmon text</H7>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should accept additional props', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <H7 accessible={false}>{'"We may speak about a place where there are no tears, no death, no fear, no night; but those are just the benefits of heaven. The beauty of heaven is seeing God." â€• Max Lucado'}</H7>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
