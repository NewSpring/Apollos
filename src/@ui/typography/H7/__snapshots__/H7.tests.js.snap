// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the H7 component should accept additional props 1`] = `
<Text
  accessible={false}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#303030",
      "fontFamily": "Colfax-Regular",
      "fontSize": 14,
      "lineHeight": 15.78,
      "paddingBottom": 1.8,
      "paddingTop": 3.6,
    }
  }
>
  "We may speak about a place where there are no tears, no death, no fear, no night; but those are just the benefits of heaven. The beauty of heaven is seeing God." â€• Max Lucado
</Text>
`;

exports[`the H7 component should accept and render passed in styles 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "salmon",
      "fontFamily": "Colfax-Regular",
      "fontSize": 14,
      "lineHeight": 15.78,
      "paddingBottom": 1.8,
      "paddingTop": 3.6,
    }
  }
>
  Salmon text
</Text>
`;

exports[`the H7 component should render correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#303030",
      "fontFamily": "Colfax-Regular",
      "fontSize": 14,
      "lineHeight": 15.78,
      "paddingBottom": 1.8,
      "paddingTop": 3.6,
    }
  }
>
  Default H7 text
</Text>
`;
